metabase/metabase
# fixes replace missing values with zeros on timeseries charts

Closes https://github.com/metabase/metabase/issues/42030

### Description

Fixes "replace missing values => zero" series setting. Previously, it was only replacing missing values for existing in the dataset x-values. However, it does not produce desired charts on time series datasets with gaps in x-values, for instace, when dataset has Janurary and March, it would be expected to see zero value for February when this setting is enabled.

### How to verify

Ensure for the following dataset when "replace missing values" is zero, it shows zeros for 2021 and 2022:
```sql
SELECT DATE '2020-01-01' AS x, 10 AS y
UNION ALL
SELECT DATE '2023-01-01', 10
UNION ALL
SELECT DATE '2024-01-01', 10
```

### Demo

<img width="1646" alt="Screenshot 2024-05-01 at 8 39 43 PM" src="https://github.com/metabase/metabase/assets/14301985/7d484185-b2a4-4e41-b69f-f0c1ac74e74d">

### Checklist
